### PR TITLE
update require/exports to import/export statements

### PR DESCRIPTION
**Pull request by roemhildtg**
Codemod: https://gist.github.com/roemhildtg/1aad302cabe76add0f29804d0c6dfb91

### CODE REVIEW

File: docs/can-canjs/can-core.md [hunk 1]
```diff
@@ -44,9 +44,7 @@ More commonly, a compute derives its value from other observables.  The followin
 compute:
 
 ```js
-var DefineMap = require("can-define/map/map"),
-	DefineList = require("can-define/list/list"),
-	compute = require("can-compute");
+import DefineMap from 'can-define/map/map';import DefineList from 'can-define/list/list';import compute from 'can-compute';
```
Comment (phillipskevin) [inline docs/can-canjs/can-core.md]: Can you put in a newline here?

File: docs/can-canjs/can-core.md [hunk 2]
```diff
@@ -183,10 +181,7 @@ to a service layer. This is often done via the
 into a single api:
 
 ```js
-var baseMap = require("can-connect/can/base-map/base-map"),
-    DefineMap = require("can-define/map/map"),
-    DefineList = require("can-define/list/list"),
-	set = require("can-set");
+import baseMap from 'can-connect/can/base-map/base-map';import DefineMap from 'can-define/map/map';import DefineList from 'can-define/list/list';import set from 'can-set';
```
Comment (phillipskevin) [inline docs/can-canjs/can-core.md]: Another newline 😄 .

File: docs/can-canjs/can-infrastructure.md [hunk 3]
```diff
@@ -298,7 +298,7 @@ var unicorn = {
 if (namespace.unicorn) {
 	throw new Error("You can't have two versions of can-unicorn, check your dependencies");
 } else {
-	module.exports = namespace.unicorn = unicorn;
+	export default namespace.unicorn = unicorn;
```
Comment (phillipskevin) [inline docs/can-canjs/can-infrastructure.md]: You can't have exports inside of if statements with es6. We'll have to figure out another way to do this.

File: docs/can-canjs/can-namespace.md [hunk 4]
```diff
@@ -24,7 +24,7 @@ var unicorn = {
 if (namespace.unicorn) {
 	throw new Error("You can’t have two versions of can-unicorn; check your dependencies");
 } else {
-	module.exports = namespace.unicorn = unicorn;
+	export default namespace.unicorn = unicorn;
```
Comment (phillipskevin) [inline docs/can-canjs/can-namespace.md]: Same as above. Can't have export inside of a block.

File: docs/can-guides/commitment/recipes/todomvc-with-steal/10-connection-list/index.js [hunk 5]
```diff
@@ -1,9 +1,8 @@
 // index.js
-var view = require("./index.stache");
-var DefineMap = require("can-define/map/");
-var Todo = require("~/models/todo");
-require("~/models/todos-fixture");
-
+import view from './index.stache';
+import DefineMap from 'can-define/map/';
+import Todo from '~/models/todo';import '~/models/todos-fixture';
```
Comment (phillipskevin) [inline docs/can-guides/commitment/recipes/todomvc-with-steal/10-connection-list/index.js]: Another newline 😄 .

File: docs/can-guides/commitment/recipes/todomvc-with-steal/15-setter-toggle/index.js [hunk 6]
```diff
@@ -1,9 +1,8 @@
 // index.js
-var view = require("./index.stache");
-var DefineMap = require("can-define/map/");
-var Todo = require("~/models/todo");
-require("~/models/todos-fixture");
-
+import view from './index.stache';
+import DefineMap from 'can-define/map/';
+import Todo from '~/models/todo';import '~/models/todos-fixture';
```
Comment (phillipskevin) [inline docs/can-guides/commitment/recipes/todomvc-with-steal/15-setter-toggle/index.js]: Another newline 😄 .

File: docs/can-guides/commitment/recipes/todomvc-with-steal/17-routing/index.js [hunk 7]
```diff
@@ -1,9 +1,8 @@
 // index.js
-var view = require("./index.stache");
-var DefineMap = require("can-define/map/");
-var Todo = require("~/models/todo");
-var route = require("can-route");
-require("~/models/todos-fixture");
+import view from './index.stache';
+import DefineMap from 'can-define/map/';
+import Todo from '~/models/todo';
+import route from 'can-route';import '~/models/todos-fixture';
```
Comment (phillipskevin) [inline docs/can-guides/commitment/recipes/todomvc-with-steal/17-routing/index.js]: Another newline 😄 .

File: docs/can-guides/commitment/recipes/todomvc-with-steal/17-routing/index.js [hunk 8]
```diff
@@ -41,6 +40,4 @@ route("{filter}");
 route.start();
 
 var frag = view(appVM);
-document.body.appendChild(frag);
-
-require("can-todomvc-test")(appVM);
+document.body.appendChild(frag);import 'can-todomvc-test';(appVM);
```
Comment (phillipskevin) [inline docs/can-guides/commitment/recipes/todomvc-with-steal/17-routing/index.js]: This `import` should be moved to the top so the function can be called correctly here. Like it's done up above in 15-setter-toggle/index.js (and some other examples probably).

File: docs/can-guides/commitment/recipes/todomvc-with-steal/4-render-todos/index.js [hunk 9]
```diff
@@ -21,6 +21,4 @@ var appVM = window.appVM = new AppViewModel({
 });
 
 var frag = view(appVM);
-document.body.appendChild(frag);
-
-require("can-todomvc-test")(appVM);
+document.body.appendChild(frag);import 'can-todomvc-test';(appVM);
```
Comment (phillipskevin) [inline docs/can-guides/commitment/recipes/todomvc-with-steal/4-render-todos/index.js]: Same thing - need to fix how this import is called.

File: docs/can-guides/commitment/recipes/todomvc-with-steal/todomvc-with-steal.md [hunk 10]
```diff
@@ -134,8 +134,7 @@ StealJS.
 - Use [can-todomvc-test](https://www.npmjs.com/package/can-todomvc-test) to load the application’s
   styles and run its tests:
 
-  ```js
-  require("can-todomvc-test")(appVM);
+  ```jsimport 'can-todomvc-test';(appVM);
```
Comment (phillipskevin) [inline docs/can-guides/commitment/recipes/todomvc-with-steal/todomvc-with-steal.md]: Need a newline here and to name the import and call it correctly.

File: docs/can-guides/commitment/recipes/weather-report/advanced.html [hunk 11]
```diff
@@ -65,10 +65,9 @@ <h1>10 day {{place.name}} Weather Forecast</h1>
   }
 </script>
 <script src="../../../../../node_modules/steal/steal.js" main="@empty">
-var can = require("can/all");
-can.defineStreamKefir = require("can-define-stream-kefir");
+import can from 'can/all';import can.defineStreamKefir from 'can-define-stream-kefir';
```
Comment (phillipskevin) [inline docs/can-guides/commitment/recipes/weather-report/advanced.html]: This one needs a newline and should be something like
```
import defineStreamKefir from 'can-define-stream-kefir';

can.defineStreamKefir = defineStreamKefir;
```

File: docs/can-guides/commitment/recipes/weather-report/advanced/start.html [hunk 12]
```diff
@@ -53,9 +53,8 @@ <h1>10 day {{place.name}} Weather Forecast</h1>
   }
 </script>
 <script src="../../../../../../node_modules/steal/steal.js" main="@empty">
-var can = require("can/all");
-can.defineStreamKefir = require("can-define-stream-kefir");
-var Kefir = require("kefir");
+import can from 'can/all';import can.defineStreamKefir from 'can-define-stream-kefir';
```
Comment (phillipskevin) [inline docs/can-guides/commitment/recipes/weather-report/advanced/start.html]:
```
import can from 'can/all';
import defineStreamKefir from 'can-define-stream-kefir';
import Kefir from 'kefir';

can.defineStreamKefir = defineStreamKefir;
```

File: docs/can-guides/commitment/recipes/weather-report/advanced/working.html [hunk 13]
```diff
@@ -65,9 +65,8 @@ <h1>10 day {{place.name}} Weather Forecast</h1>
   }
 </script>
 <script src="../../../../../../node_modules/steal/steal.js" main="@empty">
-var can = require("can/all");
-can.defineStreamKefir = require("can-define-stream-kefir");
-var Kefir = require("kefir");
+import can from 'can/all';import can.defineStreamKefir from 'can-define-stream-kefir';
```
Comment (phillipskevin) [inline docs/can-guides/commitment/recipes/weather-report/advanced/working.html]: same can.defineStreamKefir approach.

File: docs/can-guides/components/demo.html [hunk 14]
```diff
@@ -57,9 +57,8 @@
 </todos-app>
 </script>
 <script src="../../node_modules/steal/steal.js" main="@empty">
-var $ = require('jquery');
-var can = require('can');
-require('can/model/model');
+import $ from 'jquery';
+import can from 'can';import 'can/model/model';
```
Comment (phillipskevin) [inline docs/can-guides/components/demo.html]: newline here.

File: docs/can-guides/components/helpers-0.html [hunk 15]
```diff
@@ -1,8 +1,7 @@
 <div id="app"></div>
 <script src="../../node_modules/steal/steal.js" main="@empty">
-var $ = require('jquery');
-var can = require('can');
-require('can/model/model');
+import $ from 'jquery';
+import can from 'can';import 'can/model/model';
```
Comment (phillipskevin) [inline docs/can-guides/components/helpers-0.html]: newline 😄 .

File: docs/can-guides/components/meetup-start.html [hunk 16]
```diff
@@ -65,9 +65,8 @@ <h1>todos</h1>
 </section>
 
 <script src="../../node_modules/steal/steal.js" main="@empty">
-var $ = require('jquery');
-var can = require('can');
-require('can/model/model');
+import $ from 'jquery';
+import can from 'can';import 'can/model/model';
```
Comment (phillipskevin) [inline docs/can-guides/components/meetup-start.html]: newline 😄 .

File: docs/can-guides/components/meetup-working.html [hunk 17]
```diff
@@ -49,8 +49,7 @@ <h1>todos</h1>
 </footer>
 </section>
 <script src="../../node_modules/steal/steal.js" main="@empty">
-var $ = require('jquery');
-var can = require('can');
-require('can/model/model');
+import $ from 'jquery';
+import can from 'can';import 'can/model/model';
```
Comment (phillipskevin) [inline docs/can-guides/components/meetup-working.html]: newline 😄 .

File: docs/can-guides/components/scope-0.html [hunk 18]
```diff
@@ -1,8 +1,7 @@
 <div id="app"></div>
 <script src="../../node_modules/steal/steal.js" main="@empty">
-var $ = require('jquery');
-var can = require('can');
-require('can/model/model');
+import $ from 'jquery';
+import can from 'can';import 'can/model/model';
```
Comment (phillipskevin) [inline docs/can-guides/components/scope-0.html]: newline 😄 .

File: docs/can-guides/components/scope-1.html [hunk 19]
```diff
@@ -1,8 +1,7 @@
 <div id="app"></div>
 <script src="../../node_modules/steal/steal.js" main="@empty">
-var $ = require('jquery');
-var can = require('can');
-require('can/model/model');
+import $ from 'jquery';
+import can from 'can';import 'can/model/model';
```
Comment (phillipskevin) [inline docs/can-guides/components/scope-1.html]: newline 😄 .

File: docs/can-guides/components/scope-2.html [hunk 20]
```diff
@@ -1,8 +1,7 @@
 <div id="app"></div>
 <script src="../../node_modules/steal/steal.js" main="@empty">
-var $ = require('jquery');
-var can = require('can');
-require('can/model/model');
+import $ from 'jquery';
+import can from 'can';import 'can/model/model';
```
Comment (phillipskevin) [inline docs/can-guides/components/scope-2.html]: newline 😄 .

File: docs/can-guides/components/scope-3.html [hunk 21]
```diff
@@ -1,8 +1,7 @@
 <div id="app"></div>
 <script src="../../node_modules/steal/steal.js" main="@empty">
-var $ = require('jquery');
-var can = require('can');
-require('can/model/model');
+import $ from 'jquery';
+import can from 'can';import 'can/model/model';
```
Comment (phillipskevin) [inline docs/can-guides/components/scope-3.html]: newline 😄 .

File: docs/can-guides/components/template-0.html [hunk 22]
```diff
@@ -1,8 +1,7 @@
 <div id="app"></div>
 <script src="../../node_modules/steal/steal.js" main="@empty">
-var $ = require('jquery');
-var can = require('can');
-require('can/model/model');
+import $ from 'jquery';
+import can from 'can';import 'can/model/model';
```
Comment (phillipskevin) [inline docs/can-guides/components/template-0.html]: newline 😄 .

File: docs/can-guides/components/template-1.html [hunk 23]
```diff
@@ -1,8 +1,7 @@
 <div id="app"></div>
 <script src="../../node_modules/steal/steal.js" main="@empty">
-var $ = require('jquery');
-var can = require('can');
-require('can/model/model');
+import $ from 'jquery';
+import can from 'can';import 'can/model/model';
```
Comment (phillipskevin) [inline docs/can-guides/components/template-1.html]: newline 😄 .

File: docs/can-guides/components/template-2.html [hunk 24]
```diff
@@ -1,8 +1,7 @@
 <div id="app"></div>
 <script src="../../node_modules/steal/steal.js" main="@empty">
-var $ = require('jquery');
-var can = require('can');
-require('can/model/model');
+import $ from 'jquery';
+import can from 'can';import 'can/model/model';
```
Comment (phillipskevin) [inline docs/can-guides/components/template-2.html]: newline 😄 .

File: docs/can-guides/components/template.html [hunk 25]
```diff
@@ -1,9 +1,8 @@
 <div id="app"></div>
 <link rel="stylesheet" type="text/css" href="base.css"/>
 <script src="../../node_modules/steal/steal.js" main="@empty">
-var $ = require('jquery');
-var can = require('can');
-require('can/model/model');
+import $ from 'jquery';
+import can from 'can';import 'can/model/model';
```
Comment (phillipskevin) [inline docs/can-guides/components/template.html]: newline 😄 .

File: docs/can-guides/components/todomvc-alt.html [hunk 26]
```diff
@@ -59,10 +59,8 @@ <h1>todos</h1>
 <div id='app'></div>
 
 <script src="../../node_modules/steal/steal.js" main="@empty">
-var $ = require('jquery');
-var can = require('can');
-require('can/model/model');
-require('can/util/fixture/fixture');
+import $ from 'jquery';
+import can from 'can';import 'can/model/model';import 'can/util/fixture/fixture';
```
Comment (phillipskevin) [inline docs/can-guides/components/todomvc-alt.html]: newlines 😄 .

File: docs/can-guides/components/todomvc-light.html [hunk 27]
```diff
@@ -59,9 +59,8 @@ <h1>todos</h1>
 
 </script>
 <script src="../../node_modules/steal/steal.js" main="@empty">
-var $ = require('jquery');
-var can = require('can');
-require('can/model/model');
+import $ from 'jquery';
+import can from 'can';import 'can/model/model';
```
Comment (phillipskevin) [inline docs/can-guides/components/todomvc-light.html]: newline 😄 .

File: docs/can-guides/components/todomvc-small.html [hunk 28]
```diff
@@ -59,9 +59,8 @@ <h1>todos</h1>
 
 </script>
 <script src="../../node_modules/steal/steal.js" main="@empty">
-var $ = require('jquery');
-var can = require('can');
-require('can/model/model');
+import $ from 'jquery';
+import can from 'can';import 'can/model/model';
```
Comment (phillipskevin) [inline docs/can-guides/components/todomvc-small.html]: newline 😄 .

File: docs/can-guides/components/todomvc.html [hunk 29]
```diff
@@ -53,9 +53,8 @@ <h1>todos</h1>
 
 </script>
 <script src="../../node_modules/steal/steal.js" main="@empty">
-var $ = require('jquery');
-var can = require('can');
-require('can/model/model');
+import $ from 'jquery';
+import can from 'can';import 'can/model/model';
```
Comment (phillipskevin) [inline docs/can-guides/components/todomvc.html]: newline 😄 .

File: docs/can-guides/contribute/making-a-new-package.md [hunk 30]
```diff
@@ -47,7 +47,7 @@ __3.__ List `canjs` in your `package.json`’s `keywords`.
 
 __4.__ Update the code to match the [guides/contributing/project-organization#Fileorganizationandresponsibilities File organization and responsibilities] section.  There are a few changes to make:
 
-- Change everything to CommonJS.  Use `require('module-name')` instead of `import 'module-name'`.
+- Change everything to CommonJS.  Useimport ` from 'module-name';` instead of `import 'module-name'`.
```
Comment (phillipskevin) [inline docs/can-guides/contribute/making-a-new-package.md]: Let's not change this file just yet. We don't have plans to change can-* source files to ES6 modules right now.

File: docs/can-guides/experiment/debugging/debugging.md [hunk 31]
```diff
@@ -48,8 +48,7 @@ StealJS supports two ways of loading `can-debug` only in production:
 Add the following to your main module:
 
 ```js
-//!steal-remove-start
-window.can = require('can-debug');
+//!steal-remove-startimport window.can from 'can-debug';
```
Comment (phillipskevin) [inline docs/can-guides/experiment/debugging/debugging.md]:
```
//!steal-remove-start
import can from 'can-debug';
window.can = can;
```

File: docs/can-guides/experiment/debugging/debugging.md [hunk 32]
```diff
@@ -78,18 +77,17 @@ To import the debug module only in development, add the following
 code to your main module:
 
 ```js
-if (process.env.NODE_ENV !== "production") {
-  window.can = require("can-debug");
+if (process.env.NODE_ENV !== "production") {import window.can from 'can-debug';
```
Comment (phillipskevin) [inline docs/can-guides/experiment/debugging/debugging.md]: same thing. 
```
import can from 'can-debug';
window.can = can;
```

File: docs/can-guides/introduction/technical.md [hunk 33]
```diff
@@ -725,10 +725,7 @@ into individual and independent modules and files that look like:
 The __Model__, in _models/todo.js_, looks like:
 
 ```js
-var DefineMap = require("can-define/map/map"),
-    DefineList = require("can-define/list/list"),
-    set = require("can-set"),
-    superMap = require("can-connect/can/super-map/super-map");
+import DefineMap from 'can-define/map/map';import DefineList from 'can-define/list/list';import set from 'can-set';import superMap from 'can-connect/can/super-map/super-map';
```
Comment (phillipskevin) [inline docs/can-guides/introduction/technical.md]: newlines 😸 .

File: docs/can-guides/introduction/technical.md [hunk 34]
```diff
@@ -1429,11 +1426,8 @@ This example also uses [can-define/list/list] to define a type for an array of `
 Using [can-connect], we’ll create a connection between a RESTful `/api/todos` service and our `Todo` instances and `TodoList` lists:
 
 ```js
-var connect = require("can-connect");
-Todo.connection = connect([
-	require("can-connect/can/map/map"),
-	require("can-connect/constructor/constructor"),
-	require("can-connect/data/url/url")
+import connect from 'can-connect';
```
Comment (phillipskevin) [inline docs/can-guides/introduction/technical.md]: newlines 😸 .

File: docs/can-guides/introduction/technical.md [hunk 35]
```diff
@@ -1587,17 +1581,14 @@ The `{completed: false}` object is passed to the server as parameters and repres
...
 
 Todo.algebra = new set.Algebra(
 	set.props.boolean("completed")
 );
 
-Todo.connection = connect([
-	require("can-connect/can/map/map"),
-	require("can-connect/constructor/constructor"),
-	require("can-connect/data/url/url")
+Todo.connection = connect([import 'can-connect/can/map/map';import 'can-connect/constructor/constructor';import 'can-connect/data/url/url';
```
Comment (phillipskevin) [inline docs/can-guides/introduction/technical.md]: These imports need to be moved to the top and named.

File: docs/can-guides/upgrade/migrating-to-canjs-3.md [hunk 36]
```diff
@@ -572,8 +572,7 @@ If you’ve used [can-map-define] in the past, then using [can-define] should be
 A typical map looks like:
 
 ```js
-var Map = require("can-map");
-require("can-map-define");
+import Map from 'can-map';import 'can-map-define';
```
Comment (phillipskevin) [inline docs/can-guides/upgrade/migrating-to-canjs-3.md]: newline.

**Comment by roemhildtg** [comment]
Updated. Its weird that some files worked and others didn't. Line ending style perhaps?

**Comment by phillipskevin** [comment]
Great job! I'll fix up the conflicts and merge.

**Pull request merged**